spec/classes/redis_spec.rb

```diff
@@ -1478,6 +1478,146 @@ class { 'redis::globals':
+            'content' => %r{^io-threads-do-reads yes$}
+          )
+        }
+      end
+
+      describe 'test dynamic-hz for redis6' do
+        let(:params) do
+          {
+            dynamic_hz: true,
+          }
+        end
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^dynamic-hz yes$}
+          )
+        }
+      end
+
+      describe 'test activedefrag for redis6' do
+        let(:params) do
+          {
+            activedefrag: false,
+          }
+        end
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^activedefrag no$}
+          )
+        }
+      end
+
+      describe 'test jemalloc-bg-thread for redis6' do
+        let(:params) do
+          {
+            jemalloc_bg_thread: true,
+          }
+        end
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^jemalloc-bg-thread yes$}
+          )
+        }
+      end
+
+      describe 'test activedefrag configuration for redis6' do
+        let(:params) do
+          {
+            activedefrag: true,
+            active_defrag_ignore_bytes: '200mb',
+            active_defrag_threshold_lower: 11,
+            active_defrag_threshold_upper: 99,
+            active_defrag_cycle_min: 7,
+            active_defrag_cycle_max: 23,
+            active_defrag_max_scan_fields: 1341,
+          }
+        end
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^activedefrag yes$}
+          )
+        }
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^active-defrag-ignore-bytes 200mb$}
+          )
+        }
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^active-defrag-threshold-lower 11$}
+          )
+        }
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^active-defrag-threshold-upper 99$}
+          )
+        }
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^active-defrag-cycle-min 7$}
+          )
+        }
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^active-defrag-cycle-max 23$}
+          )
+        }
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^active-defrag-max-scan-fields 1341$}
+          )
+        }
+      end
+
+      describe 'test rdb-save-incremental-fsync for redis6' do
+        let(:params) do
+          {
+            rdb_save_incremental_fsync: true,
+          }
+        end
+
+        it {
+          is_expected.to contain_file(config_file_orig).with(
+            'content' => %r{^rdb-save-incremental-fsync yes$}
+          )
+        }
```

Comment:
You do not have to reformat this but I think having this in a single line is best practice:
```suggestion
        it { is_expected.to contain_file(config_file_orig).with('content' => %r{^rdb-save-incremental-fsync yes$}) }
```

Reply:
Fixed in most recent commit. Thanks :)